### DISCUSSION TITLE
Can the response body be included when using the `--json` flag?

### DISCUSSION THREAD

**Discussion by selfagency** [Orange-OpenSource/hurl#1969] [Q&A] [answered]
Are there any flags I can pass so that the response body is one of the fields within the JSON result returned when using the `--json` flag? Currently, it appears only to provide headers, cookies, timings, and other metadata for each request, but I can't tell if there's a way to get the body in there too, instead of as a separate entity outside of the JSON result.

**Comment by jcamiel** [comment, marked as answer]
Hi @selfagency currently no `--json` doesn't include response bodies. We plan do add it (see #1907), we just need to see if we include the response by default, or with an option opt-in.

**Comment by iBicha** [comment]
I advocate for having the response NOT be included by default in the JSON output, and that if specified, the `--json` would specify the "run" details, while another flag would be for the raw output. My reasoning is:
- Lots of the time the JSON without the output is enough because you already captured the needed information. In other times, we want to save both the run details and the response separately (and track these files in source control)
- The raw response can be a very big file, which reduces the visibility/usability of the JSON.
- The format of the response can simply not fit well into the JSON payload. Imagine an xml file (response) in the JSON. You can't properly look at it or format it.
  - I know this is less common, but an even extreme example where the response wouldn't fit in the JSON file, is the Triton Inference Server's [Binary Tensor Data Extension](https://github.com/triton-inference-server/server/blob/main/docs/protocol/extension_binary_data.md#binary-tensor-data-extension). The response is a hybrid of JSON and binary. That's just a mess to include in the json file. Response formats can be really varied, and committing to be able to include all of them into the json file can be a really hard task and unproductive. That's why I think a separate file would be better.
- The last reason for the separate file for the response is that it can has the right file extension, which in turn allows the formatting of the file (json, xml, etc) and that would be the responsibility of whatever formatter used, instead of the responsibility of hurl (which shouldn't be altering the body for many reasons)

**Reply by selfagency** [reply]
I don't care whether it's included by default as long as there's a way to include it. If I have to do `--json --full` or something like that, that's fine by me.
